src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -213,6 +213,20 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
   function commitWork(current : ?Fiber, finishedWork : Fiber) : void {
     switch (finishedWork.tag) {
       case ClassComponent: {
+        const instance = finishedWork.stateNode;
+        if (!current) {
+          if (typeof instance.componentDidMount === 'function') {
+            instance.componentDidMount();
+          }
+        } else {
+          if (typeof instance.componentDidUpdate === 'function') {
+            const prevProps = current.memoizedProps;
+            // TODO: This is the new state. We don't currently have the previous
+            // state anymore.
```

Comment:
We already keep the about-to-be-flushed update queue in `callbackList` so that we can call the callbacks. What if we add a `previousState` field to queue? It can be set when `mergeUpdateQueue` is called. Then rename `callbackList` to something broader.